File: js/4.js
/* 
* @Author: ben_cripps
* @Date:   2014-12-05 19:51:04
* @Last Modified by:   ben_cripps
* @Last Modified time: 2014-12-05 20:03:23
*/

'use strict';

function isNumericPalindrome(n) {
	return n === parseInt( String(n).split('').reverse().join('') );
}

function loop(n) {
	for (var i = n; i >= 0; i--) {
		if ( isNumericPalindrome( n * i ) ) return ['The palindrome is ', n * i, ', and the multiples are ', n, ' and 999'].join('');
	}
}

console.log( loop(999) )

File: python/4.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# @Author: ben_cripps
# @Date:   2015-01-02 22:46:28
# @Last Modified by:   ben_cripps
# @Last Modified time: 2015-01-02 22:55:03

def isPalindrome(n):
    return int(str(n)[::-1]) == n

numbers = [x*y for x in range(999) for y in range(999)]

print(max(filter(isPalindrome, numbers)))